libgstd/gstd_state.c

```diff
@@ -249,3 +285,21 @@ gstd_state_read (GstdState * self)
 
   return current;
 }
+
+GstdReturnCode
+gstd_state_increment_refcount (GstdState * self)
+{
+  g_return_val_if_fail (self, GSTD_NULL_ARGUMENT);
+  self->refcount++;
+  return GSTD_EOK;
+}
+
+GstdReturnCode
+gstd_state_decrement_refcount (GstdState * self)
+{
+  g_return_val_if_fail (self, GSTD_NULL_ARGUMENT);
+  if (0 < self->refcount) {
+    self->refcount--;
```

Comment:
Does it make sense to protect refcount variable with object lock/unlock here or the call protection is good enough?

Reply:
I don't think that a lock here is needed because the only way to modify these is through the calls and they are all protected. The extra lock would make it easier to extend while maintaining thread safety, but it can also cause a deadlock if it is not used correctly. For example someone calling this without locking the session before the call.